File: systems/system.py
from systems.planet import Planet

class SolarSystem:
    def __init__(self, name):
        self.planets = []
        self.ships = []
        self.name = name

    def add_planet(self, pop, wealth, x, y):
        self.planets.append(Planet(self.name, pop, wealth, x, y))

    def register_ship(self, ship):
        self.ships.append(ship)

    def unregister_ship(self, ship):
        self.ships.remove(ship)

    def update(self, dt):
        for planet in self.planets:
            planet.update(dt)

File: dashboard/src/simulation/system.js
class SolarSystem {
  static restore({ planets, ships }) {
    return new SolarSystem(planets, ships);
  }

  constructor(planets, ships) {
    this.planets = planets;
    this.ships = ships;
  }

  toJSON() {
    let { planets, ships } = this;
    return {
      planets,
      ships
    }
  }
}

class Planet {
  static restore({ system, pop, wealth, x, y }) {
    return new Planet(system, pop, wealth, x, y);
  }

  constructor(system, pop, wealth, x, y) {
    this.system = system;
    this.pop = pop;
    this.wealth = wealth;
    this.x = x;
    this.y = y;
  }

  toJSON() {
    let { system, pop, wealth, x, y } = this;
    return {
      pop,
      wealth,
      x,
      y
    }
  }
}

if(!window.simulation) {
  window.simulation = {};
}

window.simulation.systems = {
  system: {
    'SolarSystem': SolarSystem
  },
  planet: {
    'Planet': Planet
  }
};